### PR TITLE
Update PR template to include todo list for documentation-only changes

### LINKED ISSUE #1701: Update PR template to explain what to do for docs-only changes
We need to better guide people on how to submit PRs for docs-only changes. See this comment: https://github.com/BetterThanTomorrow/calva/pull/1700#issuecomment-1114000290.

### PR DESCRIPTION
**Pull request by bpringe**
## What has Changed?

- Updates the PR template to include todo list for documentation-only changes
- Updates CI to merge published to dev when changes are pushed to published

This PR is directed at `published` intentionally.

Fixes #1701

## My Calva PR Checklist

I have:

- [ ] Read [How to Contribute](https://github.com/BetterThanTomorrow/calva/wiki/How-to-Contribute#before-sending-pull-requests).
- [ ] Directed this pull request at the `dev` branch. (Or have specific reasons to target some other branch.)
- [ ] Made sure I have changed the PR base branch, so that it is not `published`. (Sorry for the nagging.)
- [ ] Updated the `[Unreleased]` entry in `CHANGELOG.md`, linking the issue(s) that the PR is addressing.
- [ ] Figured if **anything** about the fix warrants tests on Mac/Linux/Windows/Remote/Whatever, and either tested it there if so, or mentioned it in the PR.
- [ ] Added to or updated docs in this branch, if appropriate
- [ ] Tests
     - [ ] Tested the particular change
     - [ ] Figured if the change might have some side effects and tested those as well.
     - [ ] Smoke tested the extension as such.
     - [ ] Tested the VSIX built from the PR (so, after you've submitted the PR). You'll find the artifacts by clicking _Show all checks_ in the CI section of the PR page, and then _Details_ on the `ci/circleci: build` test.
- [ ] Referenced the issue I am fixing/addressing _in a commit message for the pull request_.
     - [ ] If I am fixing the issue, I have used [GitHub's fixes/closes syntax](https://help.github.com/en/articles/closing-issues-using-keywords)
     - [ ] If I am fixing just part of the issue, I have just referenced it w/o any of the "fixes” keywords.
- [ ] Created the issue I am fixing/addressing, if it was not present.
- [ ] Formatted all JavaScript and TypeScript code that was changed. (use the [prettier extension](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode) or run `npm run prettier-format`)
- [ ] Confirmed that there are no linter warnings or errors (use the [eslint extension](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint), run `npm run eslint` before creating your PR, or run `npm run eslint-watch` to eslint as you go).

Ping @pez, @bpringe, @corasaurus-hex, @Cyrik

### CODE REVIEW

**Comment by PEZ** [comment]
I want to challenge (again) why we direct doc PRs to `published`. I constantly run into troubles with this. It would be great if these too could be directed to `dev` and then if they are urgent we can cherry pick to `published`. I'm probably missing something, but anyway, would be great if it could work.

**Comment by bpringe** [comment]
Hmm, we have already been directing people to make documentation-only changes on published in https://github.com/BetterThanTomorrow/calva/wiki/How-to-Hack-on-Calva (probably my doing in the past), and I just updated it a bit more to mention that in another place.

What troubles do you run into with this? I just merged a readme change to published, and then I merged published to dev. I don't ever run into problems with this.

It's ideal for changes to documentation only to go directly to published so we don't need to wait on the next merge of dev to published or cherry pick.

**Comment by bpringe** [comment]
The only thing other thing I see that would be good for us to do with this approach is to update the CI to merge published to dev when an update is made to published, which we discussed before.

**Comment by bpringe** [comment]
> The only thing other thing I see that would be good for us to do with this approach is to update the CI to merge published to dev when an update is made to published, which we discussed before.

If we did go with the approach in this PR, maybe we should implement the above in CI first, so no one is editing old docs on dev if we forget to manually merge published to dev. 🤔 

I just don't like docs-only changes going to dev. They should just to to published and be made immediately available to users on the site/readme. So we should work out any issues that exist, if possible.

**Comment by PEZ** [comment]
> Hmm, we have already been directing people to make documentation-only changes on published in

I know. That's what I am challenging.

> What troubles do you run into with this? I just merged a readme change to published, and then I merged published to dev. I don't ever run into problems with this.

I've never tried to merge. Thought you had adviced me to cherry pick, and some PRs have a lot of commits. It has not occurred to me that merging published into dev would be safe, but if you say it is, then that's great.

> I just don't like docs-only changes going to dev. They should just to to published and be made immediately available to users on the site/readme. So we should work out any issues that exist, if possible.

I think that it is often not that much of a hurry, and that if it is, there's nothing stopping us from cherry picking it.

Take a look at the current docs PR. It's bumping the version and updating package.json. Probably because the changes where made on `dev` then the PR  directed to `published`. If we should keep directing docs and code changes differently we probably need to start earlier with instructions on which branch to actually do the changes on.

File: docs/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -34,7 +35,15 @@ Fixes #
...
 <!-- Strike out (using `~`) items that do not apply, as Github reports how many are not ticked. If you want to add checkboxes, please do. -->
 
-I have:
+If this PR involves only documentation changes, I have:
+
+- [ ] Read [Editing Documentation](https://github.com/BetterThanTomorrow/calva/wiki/How-to-Hack-on-Calva#editing-documentation)
+- [ ] Directed this pull request at the `published` branch.
+- [ ] Referenced the issue I am fixing/addressing _in a commit message for the pull request_ (if there was is an issue for the documentation change)
+  - [ ] If I am fixing the issue, I have used [GitHub's fixes/closes syntax](https://help.github.com/en/articles/closing-issues-using-keywords)
+  - [ ] If I am fixing just part of the issue, I have just referenced it w/o any of the "fixes” keywords.
```
Comment (PEZ) [inline docs/PULL_REQUEST_TEMPLATE.md]: I think a check like *Built the docs site and tested it locally*. should be added here. Probably guarded with that this is not needed for simple typo fixes.

File: docs/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -43,17 +52,17 @@ I have:
 - [ ] Figured if **anything** about the fix warrants tests on Mac/Linux/Windows/Remote/Whatever, and either tested it there if so, or mentioned it in the PR.
 - [ ] Added to or updated docs in this branch, if appropriate
 - [ ] Tests
-     - [ ] Tested the particular change
-     - [ ] Figured if the change might have some side effects and tested those as well.
-     - [ ] Smoke tested the extension as such.
-     - [ ] Tested the VSIX built from the PR (so, after you've submitted the PR). You'll find the artifacts by clicking _Show all checks_ in the CI section of the PR page, and then _Details_ on the `ci/circleci: build` test.
```
Comment (PEZ) [inline docs/PULL_REQUEST_TEMPLATE.md:49]: Did you intend to include whitespace changes?

Comment (bpringe) [reply docs/PULL_REQUEST_TEMPLATE.md:49]: Yes, I fixed it up according to the markdownlint.

**Comment by bpringe** [comment]
> Take a look at the current docs PR. It's bumping the version and updating package.json.

Ugh, I missed this and merged it earlier (not sure if you noticed). Will that cause an issue? I don't think it will, though that version is not released yet. Later when it is though, I think the merge of dev to published should be fine.

> I've never tried to merge. Thought you had adviced me to cherry pick, and some PRs have a lot of commits. It has not occurred to me that merging published into dev would be safe, but if you say it is, then that's great.

Yes, I believe we discussed before that merging versus cherry picking was safe. But maybe I'm misremembering.

**Comment by bpringe** [comment]
Whatever changes are on published should be safe to be merged into dev, I think (and it seems to have been safe every time I've done it). dev should always be published + some.

**Comment by PEZ** [comment]
You have probably told me. I just need to actually practice things for it to stick. 😂

**Comment by bpringe** [comment]
@PEZ With the change to CI when we push to published it should now be merged into dev. I think this will also happen when we merged dev to published though, right? It's not really an issue, but is unnecessary. Do you think we can and should prevent that, though?

Anyway, this is an easy change to revert if we decide we don't want it later. I don't know how to test this really until we merge it to published. I say we merge it and see:

1. How it goes once we merge it, as the new CI step should run to merge published into dev
2. How it goes over time. If we have any trouble with it

If you agree with merging, please let me know and I will merge it myself so I can monitor the new CI step.

File: .circleci/config.yml [hunk 3]
```diff
@@ -327,29 +327,45 @@ jobs:
...
       - image: cimg/node:15.8
     steps:
       - add_ssh_keys:
           fingerprints:
-            - "d1:06:ff:c3:58:59:68:6e:a3:2f:69:1d:e2:94:7c:14"
+            - 'd1:06:ff:c3:58:59:68:6e:a3:2f:69:1d:e2:94:7c:14'
       - checkout
       - run:
-          name: Merge dev into published
+          name: Merge dev to published
```
Comment (PEZ) [inline .circleci/config.yml]: Isn't it strange to use `to` here when we use `into` everywhere else?

**Comment by PEZ** [comment]
I'm good with testing this.

**Pull request merged**